File: fix_grid_props.py
#!/usr/bin/env python3
import re
import os

files = [
    '/home/lidruf/TRANSPORT/frontend/src/pages/Dashboard.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/LandingPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/TripsPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/TicketsPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/ParcelsPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/PaymentsPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/EmployeesPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/CitiesPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/ReportsPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/pages/SettingsPage.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/components/GovernmentFooter.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/components/AdvancedFilters.tsx',
    '/home/lidruf/TRANSPORT/frontend/src/components/StatisticsChart.tsx'
]

def fix_grid_props(content):
    """Convert Grid item props from xs, sm, md to size={{xs, sm, md}}"""
    # Pattern: <Grid item xs={value} sm={value} md={value} ... >
    # Convert to: <Grid item size={{ xs: value, sm: value, md: value }} ...>
    
    # This regex matches Grid item opening tags with breakpoint props
    pattern = r'<Grid\s+item\s+([^>]*?(?:xs|sm|md|lg|xl)=\{\d+(?:\.\d+)?\}[^>]*)>'
    
    def replace_grid_item(match):
        full_text = match.group(1)
        
        # Extract all breakpoint props and their values
        breakpoints = {}
        for bp in ['xs', 'sm', 'md', 'lg', 'xl']:
            bp_pattern = rf'{bp}=\{{(\d+(?:\.\d+)?)\}}'
            bp_match = re.search(bp_pattern, full_text)
            if bp_match:
                breakpoints[bp] = bp_match.group(1)
        
        # Extract other props (everything that's not xs, sm, md, lg, xl)
        other_props = re.sub(r'\s*(xs|sm|md|lg|xl)=\{\d+(?:\.\d+)?\}\s*', ' ', full_text).strip()
        
        # Build size object
        if breakpoints:
            size_pairs = ', '.join([f'{bp}: {val}' for bp, val in breakpoints.items()])
            size_str = f'size={{ {size_pairs} }}'
            
            if other_props:
                return f'<Grid item {size_str} {other_props}>'
            else:
                return f'<Grid item {size_str}>'
        else:
            return match.group(0)
    
    return re.sub(pattern, replace_grid_item, content)

for file_path in files:
    if os.path.exists(file_path):
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        new_content = fix_grid_props(content)
        
        if content != new_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f'✓ Fixed: {os.path.basename(file_path)}')
        else:
            print(f'- No changes: {os.path.basename(file_path)}')
    else:
        print(f'✗ Not found: {file_path}')

print('\nDone!')

File: fix-grid-props.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const glob = require('glob');

/**
 * Convertit les props Grid xs, sm, md, etc. vers le format MUI v7
 * xs={12} sm={6} md={3} => size={{ xs: 12, sm: 6, md: 3 }}
 */
function fixGridProps(content) {
  // Regex pour trouver les Grid items avec xs, sm, md, lg, xl
  const gridItemRegex = /<Grid\s+item\s+([^>]*?)>/g;
  
  return content.replace(gridItemRegex, (match, props) => {
    // Extraire tous les props individuels
    const propsArray = props.trim().split(/\s+(?=[a-zA-Z])/);
    
    const sizeProps = {};
    const otherProps = [];
    
    for (const prop of propsArray) {
      if (prop.match(/^(xs|sm|md|lg|xl)=/)) {
        const [key, value] = prop.split('=');
        sizeProps[key] = value.replace(/[{}]/g, '');
      } else {
        otherProps.push(prop);
      }
    }
    
    // Si des props de taille ont été trouvées
    if (Object.keys(sizeProps).length > 0) {
      const sizeString = Object.entries(sizeProps)
        .map(([k, v]) => `${k}: ${v}`)
        .join(', ');
      
      const newProps = [
        ...otherProps,
        `size={{ ${sizeString} }}`
      ].filter(p => p.trim()).join(' ');
      
      return `<Grid item ${newProps}>`;
    }
    
    return match;
  });
}

// Fichiers à traiter
const files = [
  '/home/lidruf/TRANSPORT/frontend/src/pages/Dashboard.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/LandingPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/TripsPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/TicketsPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/ParcelsPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/PaymentsPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/EmployeesPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/CitiesPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/ReportsPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/pages/SettingsPage.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/components/GovernmentFooter.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/components/AdvancedFilters.tsx',
  '/home/lidruf/TRANSPORT/frontend/src/components/StatisticsChart.tsx'
];

console.log('Fixing Grid props in MUI files...\n');

files.forEach(filePath => {
  if (fs.existsSync(filePath)) {
    const content = fs.readFileSync(filePath, 'utf8');
    const fixed = fixGridProps(content);
    
    if (content !== fixed) {
      fs.writeFileSync(filePath, fixed, 'utf8');
      console.log(`✓ Fixed: ${path.basename(filePath)}`);
    } else {
      console.log(`- No changes needed: ${path.basename(filePath)}`);
    }
  } else {
    console.log(`✗ File not found: ${filePath}`);
  }
});

console.log('\nDone!');
